packages/cloud-config/oem/06_recovery.yaml

```diff
@@ -9,11 +9,19 @@ name: "Recovery partition boot setup"
 stages:
    boot:
      - name: "Recovery"
+       if: |
+            [ -n "$(blkid -L COS_SYSTEM || true)" ] || cat /proc/cmdline | grep -q "COS_RECOVERY"
+       hostname: "cos-recovery"
        commands:
        - |
             source /etc/os-release
-            if [ -n "$(blkid -L COS_SYSTEM || true)" ]; then
-                echo >> /etc/issue
-                echo "You are booting from recovery mode. Run 'cos-reset' to reset the system to $VERSION" >> /etc/issue
-                echo >> /etc/issue
-            fi
+            echo >> /etc/issue
+            echo "You are booting from recovery mode. Run 'cos-reset' to reset the system to $VERSION" >> /etc/issue
+            echo " or CURRENT=active.img cos-upgrade to upgrade the active partition" >> /etc/issue
+            echo >> /etc/issue
+       files:
+       - path: /run/cos/recovery_mode
```

Comment:
I just added this to have a canonical way to detect if we are in recovery_mode or not